File: app/src/main/java/com/suishi/live/app/ui/activity/MainActivity.java
package com.suishi.live.app.ui.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.suishi.live.app.R;
import com.swbyte.chat.runtimepermissions.PermissionsManager;
import com.swbyte.chat.runtimepermissions.PermissionsResultAction;


public class MainActivity extends Activity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        GridView grid = findViewById(R.id.grid);
        grid.setAdapter(new HoloTilesAdapter());
        PermissionsManager.getInstance().requestAllManifestPermissionsIfNecessary(this, new PermissionsResultAction() {
            @Override
            public void onGranted() {

            }

            @Override
            public void onDenied(String permission) {

            }
        });
    }



    public class HoloTilesAdapter extends BaseAdapter {

        private static final int TILES_COUNT = 2;

        private final int[] DRAWABLES = {
                R.drawable.blue_tile,
                R.drawable.green_tile,
        };

        @Override
        public int getCount() {
            return TILES_COUNT;
        }

        @Override
        public Object getItem(int position) {
            return null;
        }

        @Override
        public long getItemId(int position) {
            return position;
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {

            RelativeLayout v;
            if (convertView == null) {
                v = (RelativeLayout) getLayoutInflater().inflate(R.layout.grid_item, parent, false);
            } else {
                v = (RelativeLayout) convertView;
            }
            v.setBackgroundResource(DRAWABLES[position % 2]);

            TextView textView1 = v.findViewById(R.id.textView1);
            TextView textView2 = v.findViewById(R.id.textView2);

            String string1 = "", string2 = "";
            if (position == 0) {
                string1 = "Portrait";
                string2 = "Flv + Local";
            } else if (position == 1) {
                string1 = "Landscape";
                string2 = "Rtmp";
            } else if (position == 2) {
                string1 = "Portrait";
                string2 = "Part";
            } else if (position == 3) {
                string1 = "Portrait";
                string2 = "Screen + Rtmp";
            }
            textView1.setText(string1);
            textView2.setText(string2);

            final int currentPosition = position;
            v.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (currentPosition == 0) {
                        goPortraitAndLocal();
                    } else if (currentPosition == 1) {
                        goPart();
                    } else if (currentPosition == 2) {

                    } else if (currentPosition == 3) {
                    }
                }
            });
            return v;
        }
    }

    private void goPortraitAndLocal() {
        Intent intent = new Intent(this, VideoActivity.class);
        startActivity(intent);
    }

    private void goPart() {
        Intent intent = new Intent(this, PortraitActivity.class);
        startActivity(intent);
    }

}

File: app/src/main/java/com/suishi/live/app/ui/activity/MainActivity.kt
package com.suishi.live.app.ui.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.GridView
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.suishi.live.app.R
import com.yanzhenjie.permission.AndPermission
import com.yanzhenjie.permission.runtime.Permission

/**
 *
 */
class MainActivity : AppCompatActivity() {

    companion object{
        fun startActivity(context: Context){
            val intent=Intent(context,MainActivity::class.java)
            context.startActivity(intent)
        }
    }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val grid = findViewById<GridView>(R.id.grid)

        AndPermission.with(this)
                .runtime()
                .permission(Permission.Group.CAMERA, Permission.Group.MICROPHONE, Permission.Group.STORAGE)
                .onGranted { permissions: List<String?>? ->
                    grid.adapter = HoloTilesAdapter()
                }
                .onDenied { permissions: List<String?>? ->
                    // Storage permission are not allowed.
                    finish()
                }
                .start()
    }

    inner class HoloTilesAdapter : BaseAdapter() {
        private val DRAWABLES = intArrayOf(
                R.drawable.blue_tile,
                R.drawable.green_tile,
                R.drawable.blue_tile
        )


        override fun getCount(): Int {
            return DRAWABLES.size
        }

        override fun getItem(position: Int): Any {
            return position
        }

        override fun getItemId(position: Int): Long {
            return position.toLong()
        }

        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
            val v: RelativeLayout
            v = if (convertView == null) {
                layoutInflater.inflate(R.layout.grid_item, parent, false) as RelativeLayout
            } else {
                convertView as RelativeLayout
            }
            v.setBackgroundResource(DRAWABLES[position % 2])
            val textView1 = v.findViewById<TextView>(R.id.textView1)
            val textView2 = v.findViewById<TextView>(R.id.textView2)
            var string1 = ""
            var string2 = ""
            if (position == 0) {
                string1 = "视频播放"
                string2 = "Flv + Local"
            } else if (position == 1) {
                string1 = "推流"
                string2 = "Rtmp"
            } else if (position == 2) {
                string1 = "录制"
                string2 = "Part"
            }
            textView1.text = string1
            textView2.text = string2
            v.setOnClickListener {
                if (position == 0) {
                    VideoActivity.startActivity(this@MainActivity)
                } else if (position == 1) {
                    PortraitActivity.startActivity(this@MainActivity)
                } else if (position == 2) {
                   CameraActivity.startActivity(this@MainActivity)
                   // GLSurfaceCamera2Activity.startActivity(this@MainActivity)
                }
            }
            return v
        }
    }
}